pkgs/tools/misc/scriptcommunicator/default.nix

```diff
@@ -0,0 +1,93 @@
+{ stdenv
+, fetchFromGitHub, qmake, pkgconfig
+, libpng, zlib, unzip
+, qt5, qttools, qtserialport, qtscript, qtmultimedia
+, makeWrapper, makeDesktopItem
+}:
+
+let
+  pname = "scriptcommunicator";
+  version = "05.10";
+
+  desktopItem = makeDesktopItem {
+    name = "scriptcommunicator";
+    exec = "${pname}";
+    icon = "$out/share/scriptcommunicator/images/main_32x32.png";
+    comment = "Scriptable data terminal"; 
+    desktopName = "scriptcommunicator";
+    genericName = "scriptcommunicator";
+    categories = "Network;";
+  };
+
+in stdenv.mkDerivation rec {
+
+  inherit pname version;
+
+  nativeBuildInputs = [ unzip qmake makeWrapper ];
+  buildInputs = [  qtserialport qtscript qtmultimedia ];
+
+  # we want QtDesigner available to create UI
+  propagatedBuildInputs = [ qttools ]; 
+
+  src = fetchFromGitHub {
+    owner = "szieke";
+    repo = "ScriptCommunicator_serial-terminal";
+    rev = "Release_05_10";
+    sha256 = "0v6lh5di5pxkqklp3iyvdqmr7w66c7xdd1r8lkk43sh7w7jnbvc9";
+  };
+
+  sourceRoot = "source/ScriptCommunicator";
+
+  buildPhase = ''
+    cd DeleteFolder/DeleteFolder
+    qmake DeleteFolder.pro
+    make
+    cd ../..
+
+    cd ScriptEditor
+    qmake ScriptEditor.pro
+    make
+    cd ..
+
+    qmake ScriptCommunicator.pro
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/{bin,share/scriptcommunicator,lib}
+
+    # Need to fix this - i don't populate with it for now.
+    #cp aardvarkI2cSpi/aardvark.so cheetahSpi/cheetah.so $out/lib/
+    cp -a ScriptEditor/apiFiles exampleScripts images templates test $out/share/scriptcommunicator
+    cp ScriptEditor/ScriptEditor $out/share/scriptcommunicator/
+    cp DeleteFolder/DeleteFolder/DeleteFolder $out/share/scriptcommunicator/
+    cp ScriptCommunicator $out/share/scriptcommunicator/
+    cp documentation/Manual_ScriptCommunicator.pdf $out/share/scriptcommunicator/
+    
+    makeWrapper $out/share/scriptcommunicator/ScriptCommunicator $out/bin/scriptcommunicator \
```

Comment:
Why not copy directly to `$out/bin` and wrap with `wrapProgram $out/bin/Scriptcommunicator --[args]`?